diesel/src/insertable.rs

```diff
@@ -193,31 +188,50 @@ where
     }
 }
 
-impl<'a, T, Tab, DB> InsertValues<Tab, DB> for &'a [T]
+#[derive(Debug, Clone, Copy)]
+pub struct BatchInsert<'a, T: 'a, Tab> {
+    records: &'a [T],
+    _marker: PhantomData<Tab>,
+}
+
+impl<'a, T, Tab> IntoIterator for BatchInsert<'a, T, Tab> {
```

Comment:
Why is this impl needed?
You use `records.iter()` below

Reply:
Need to otherwise access `records` in another file and this felt cleaner than `.records().records`